packages/components/select/src/select.vue

```diff
@@ -174,11 +174,7 @@
             ref="reference"
             v-model="selectedLabel"
             type="text"
-            :placeholder="
-              typeof currentPlaceholder === 'function'
-                ? currentPlaceholder()
-                : currentPlaceholder
-            "
+            :placeholder="currentPlaceholder || t('el.select.placeholder')"
```

Comment:
It seems this doesn't solve the bug that i18n is not reactive in the `select` component. (#11502) A computed propety may be required.